### ISSUE TITLE
New rule proposal for disallow the usage of export * from when there is another named export

### ISSUE THREAD

**Issue by xianshenglu** [eslint/eslint#11846]
**Please describe what the rule should do:**
- Disallow the usage of `export * from` when there is another named export. For example, `export *`, `export const`, etc.
- Not recommend the usage of `...` in the export object. For example, `export default {...obj1,...obj2}`

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// user.js
export const getUserInfo = {
  method: 'POST',
  url: 'api/user/getUserInfo'
}
export default {
  method: 'POST',
  url: 'api/user/getUserInfo'
}

// utilities.js
export const getUserInfo = {
  method: 'POST',
  url: 'api/utilities/getUserInfo'
}
export default {
  method: 'POST',
  url: 'api/utilities/getUserInfo'
}

// indexV1.js
export * from './user.js' //user wins
// Disallow the usage of named export under code above. For example, export *, export const, etc.
export * from './utilities.js'

// indexV2.js
import * as user from './user'
import * as utilities from './utilities' //utilities wins
// Disallow the usage of ... in the export object
export default { ...user, ...utilities }

// index.js
import * as api1 from './api/indexV1.js'
import api2 from './api/indexV2.js'
// api/user/getUserInfo, api/utilities/getUserInfo
console.log(api1.getUserInfo.url, api2.getUserInfo.url)
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It's a potential bug of es.

**Are you willing to submit a pull request to implement this rule?**

If I have time
